compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUPasses.h

```diff
@@ -18,6 +18,8 @@
 namespace mlir {
 namespace iree_compiler {
 
+class TilingConfig;
+
```

Comment:
I've never seen that rule being followed in MLIR/LLVM, quite strongly the opposite? I've seen fw declarations even with templates, etc. compile time is really a big deal. Are we following a different policy in IREE?

Reply:
I think we don't have strong rules in MLIR/LLVM. It's common when there are cyclic deps. These files are complied together under `LLVMCPU` target, so I think it does not impact compilation time a lot -- if it is a pure header file. One question, why do we keep the implementation in the `.h` file but not `.cpp` file?

(I might have to take the statement back because the implementation is located in the header file.)

Reply:
Sorry, not a reason to keep things in .h. Moving the larger functions to the .cpp